### PR TITLE
[SPT-7343] Update pendulum / six to be in parity with swimbundle_utils

### PR DESCRIPTION
**Pull request by zbartl**

### CODE REVIEW

**Comment by zbartl** [comment]
@dmikov updated `pendulum` to `1.2.5` to be in parity with https://github.com/swimlane/swimbundle-utils/blob/master/requirements.txt and noticed an error on the `six` dep as well

you had mentioned needing to update a few references, but I'm not seeing anything at a glance locally / via the tests, was that for a 2.x release of `pendulum`?

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 cachetools>=2.0,<2.1
 certifi>=2017
-pendulum>=1.2.3,<1.3
+pendulum==1.2.5
```
Comment (dmikov) [inline requirements.txt:3]: the latest pendulum is 2.1.2we want to upgrade to the latest, locking swimlane to old doesn't help when other libraries install latest one. That is how we got into problems.

Comment (zbartl) [reply requirements.txt]: updated

**Comment by zbartl** [comment]
@dmikov @cjkoral updated to 2.1.2, think I've got all the breaking changes addressed but let me know if I missed anything

also looks like pendulum 2.1.2 doesn't support Python 3.4, only >=3.5 for the 3 line, assuming we're fine to remove the 3.4 step from the Travis file?

**Comment by dmikov** [comment]
I don't see any reason to support 3.4

File: swimlane/core/client.py [hunk 2]
```diff
@@ -326,7 +326,7 @@ def __call__(self, request):
 class SwimlaneJwtAuth(SwimlaneResolver):
     """Handles authentication for all requests"""
 
-    _token_expiration_buffer = pendulum.Interval(minutes=5)
+    _token_expiration_buffer = pendulum.Duration(minutes=5)
```
Comment (dmikov) [inline swimlane/core/client.py:329]: This one was the issue, so this is fixed.

**Pull request merged**